downstairs/src/region.rs

```diff
@@ -1057,17 +1163,33 @@ impl Extent {
         }
         tx.commit()?;
 
+        // Buffer writes for fewer syscalls. The 65536 buffer size here is
+        // chosen somewhat arbitrarily. We could reimplement this ourselves
+        // without BufWriter but I don't see a reason to at the moment.
+        let mut buffered_writer = BufWriter::with_capacity(65536, &inner.file);
+
+        let mut next_block_in_run = u64::MAX;
         for write in writes {
-            if writes_to_skip.contains(&write.offset.value) {
-                assert!(only_write_unwritten);
+            let block = write.offset.value;
+            if writes_to_skip.contains(&block) {
+                debug_assert!(only_write_unwritten);
                 continue;
             }
-            let byte_offset = write.offset.value * self.block_size;
 
-            inner.file.seek(SeekFrom::Start(byte_offset))?;
-            inner.file.write_all(&write.data)?;
+            // Seeking flushes the buffer so only do it when we need to
```

Comment:
Do you mean this flushes to the file, or just performs the actual writes?

Reply:
seek just performs the writes. but per the later comment, i'll rewrite to do my own buffering